kong/plugins/key-auth/handler.lua

```diff
@@ -25,6 +25,11 @@ local function load_credential(key)
     return nil, err
   end
 
+  if cred.ttl == 0 then
+    cred = { err = "credential ttl is 0" }
+    return cred, err
```

Comment:
This is pretty strange. Can't we in this case just return:
```lua
return cred, nil, 0.001
```

**without** doing:
```
cred = { err = "credential ttl is 0" }
```

Reply:
The problem is the `cred.ttl = 0`. We can return such `cred` as that would make mlcache never expires.

`cred.ttl = 0` means the data in DB is expired. Therefore, we can return TTL as `nil` instead of `0.0.01`.

Reply:
@outsinre wouldn't the `< 0`, e.g. `-2` mean that it has expired?

Reply:
From the [readme](https://github.com/thibaultcha/lua-resty-mlcache#get):

```
If returned as `ttl < 0`, `value` will be returned by get(),
    --      but not cached. This return value will be ignored if not a number
```